pkg/urbit/jets/f/ut_nest.c

```diff
@@ -3,405 +3,36 @@
-          }
-          else return _nest_sint(van, sut, tel, ref, seg, reg, gil);
-        }
-      }
-      case c3__hint:
-      case c3__face: {
-        if ( (c3n == u3r_trel(sut, 0, &p_sut, &q_sut)) ) {
-          return u3m_bail(c3__fail);
-        } else {
-          return _nest_dext(van, q_sut, tel, ref, seg, reg, gil);
-        }
-      }
-      case c3__fork: p_sut = u3t(sut);
-      {
-        if ( c3n == u3du(ref) ) switch ( ref ) {
-          default: return _nest_sint(van, sut, tel, ref, seg, reg, gil);
-
-          case c3__noun:
-            break;
-        }
-        else switch ( u3h(ref) ) {
-          default: return _nest_sint(van, sut, tel, ref, seg, reg, gil);
-
-          case c3__atom:
-          case c3__cell:
-          case c3__core:
-            break;
-        }
-
-        {
-          u3_noun dey = u3qdi_tap(p_sut);
-          u3_noun yed = dey;
-
-          while ( u3_nul != yed ) {
-            if ( c3y == _nest_dext(van, u3h(yed), c3n, ref, seg, reg, gil) ) {
-              u3z(dey);
-              return c3y;
-            } else {
-              yed = u3t(yed);
-            }
-          }
-          u3z(dey);
-          return c3n;
-        }
-      }
-      case c3__hold:
-      {
-        if ( c3y == u3qdi_has(seg, sut) ) {
-          return c3n;
-        }
-        else {
-          u3_noun hud = u3nc(u3k(sut), u3k(ref));
-
-          if ( (c3y == u3qdi_has(gil, hud)) ) {
-            u3z(hud);
-
-            return c3y;
-          }
-          else {
-            u3_noun gus = u3qdi_put(seg, sut);
-            u3_noun zoc = u3qdi_put(gil, hud);
-            u3_noun fop = u3qfu_repo(van, sut);
-            u3_noun hiv = _nest_dext(van, fop, tel, ref, gus, reg, zoc);
-
-            u3z(hud);
-            u3z(fop);
-            u3z(zoc);
-            u3z(gus);
-
-            return hiv;
-          }
-        }
-      }
-    }
-  }
-
-  static u3_noun
-  _nest_dext_to(u3_noun van,
-                u3_noun sut,
-                u3_noun tel,
-                u3_noun ref,
-                u3_noun seg,
-                u3_noun reg,
-                u3_noun gil)
-  {
-    u3_noun tyn = _nest_dext_in(van, sut, tel, ref, seg, reg, gil);
-
-    if ( (c3y == tyn) || (c3n == tel) ) {
-      return tyn;
+      return u3m_bail(c3__fail);
     } else {
-#if 0
-      u3_noun dun = u3qfu_dunq(van, "need", sut);
-      u3_noun niz = u3qfu_dunq(van, "have", ref);
-
-      u3t_push(u3nc(c3__mean, dun));
-      u3t_push(u3nc(c3__mean, niz));
-#endif
-      return u3m_error("nest-fail");
-    }
-  }
-
-  static u3_noun
-  _nest_dext(u3_noun van,
-             u3_noun sut,
-             u3_noun tel,
-             u3_noun ref,
-             u3_noun seg,
-             u3_noun reg,
-             u3_noun gil)
-  {
-
-    if ( (c3y == u3r_sing(sut, ref)) ) {
-      return c3y;
-    }
-
-    {
+      // XX Maybe it'd be nicer to use a different mote than c3__nest
+      // here. Meh.
```

Comment:
Indeed, please add `c3__dext` (maybe shifted) to `fun_m`. A little more specificity in the cache keys can't hurt.